internal/grpc/services/gateway/storageprovider.go

```diff
@@ -107,171 +103,30 @@ func (s *svc) GetHome(ctx context.Context, _ *provider.GetHomeRequest) (*provide
 	}, nil
 }
 
-func (s *svc) getHome(_ context.Context) string {
-	// TODO(labkode): issue #601, /home will be hardcoded.
-	return "/home"
+func (s *svc) getHome(ctx context.Context) string {
+	u := appctx.ContextMustGetUser(ctx)
+	return templates.WithUser(u, s.c.HomeLayout)
 }
 
 func (s *svc) InitiateFileDownload(ctx context.Context, req *provider.InitiateFileDownloadRequest) (*gateway.InitiateFileDownloadResponse, error) {
-	log := appctx.GetLogger(ctx)
-
 	if utils.IsRelativeReference(req.Ref) {
 		return s.initiateFileDownload(ctx, req)
 	}
 
-	p, st := s.getPath(ctx, req.Ref)
-	if st.Code != rpc.Code_CODE_OK {
-		return &gateway.InitiateFileDownloadResponse{
-			Status: st,
-		}, nil
-	}
-
-	if !s.inSharedFolder(ctx, p) {
-		statReq := &provider.StatRequest{Ref: req.Ref}
-		statRes, err := s.stat(ctx, statReq)
-		if err != nil {
-			return &gateway.InitiateFileDownloadResponse{
-				Status: status.NewInternal(ctx, err, "gateway: error stating ref:"+statReq.Ref.String()),
-			}, nil
-		}
-		if statRes.Status.Code != rpc.Code_CODE_OK {
-			return &gateway.InitiateFileDownloadResponse{
-				Status: statRes.Status,
-			}, nil
-		}
-		return s.initiateFileDownload(ctx, req)
-	}
-
-	if s.isSharedFolder(ctx, p) {
-		log.Debug().Str("path", p).Msg("path points to shared folder")
-		err := errtypes.PermissionDenied("gateway: cannot download share folder: path=" + p)
-		log.Err(err).Msg("gateway: error downloading")
+	statReq := &provider.StatRequest{Ref: req.Ref}
+	statRes, err := s.stat(ctx, statReq)
+	if err != nil {
 		return &gateway.InitiateFileDownloadResponse{
```

Comment:
Let's maybe also log this?

Reply:
Actually this Stat is redundant, because in `initiateFileDownload` we already call `s.find()`!